### PR TITLE
Feature #13789: improving performances into blog loading data.

### PR DESCRIPTION
**Pull request by SilverYoCha**
Linked to PR https://github.com/Silverpeas/Silverpeas-Core/pull/1319

### CODE REVIEW

File: blog/blog-library/src/main/java/org/silverpeas/components/blog/service/DefaultBlogService.java [hunk 1]
```diff
@@ -296,191 +293,146 @@ private Connection openConnection() {
...
-          if (publication.getPK()
-              .getId()
-              .equals(pubId)) {
-            posts.add(getPost(publication));
-          }
-        }
-      }
-      return posts;
+      // search publications sorted by event date
+      Collection<String> lastEventIds = PostDAO.getAllEvents(con, instanceId);
```
Comment (mmoqui) [inline blog/blog-library/src/main/java/org/silverpeas/components/blog/service/DefaultBlogService.java:338]: Pfff, this method name was originally misnamed. In fact, what this method does is to get all the blog posts of the specified component instance ordered by their event date. So the method should be renamed to something like `getAllPostsOrderedByEventDate`.
And then the variable shouldn't be named lastEventIds but something like `postsOrderedByEventDate`.

My remark is also for all similar snippet of code written in the next methods below (this allow me to not repeat this comment)

**Review by mmoqui — CHANGES_REQUESTED** [review]
I ask for renaming some original method names. It is for a better understanding for both new developers and for those who get back to the code months or years later. We have to establish a clear distinction between a blog post and an event that can be related by such a post; a post isn't an event, it creates an event (in the calendar of the blog): either the event about which its content is or the event of the post writing itself.

File: blog/blog-library/src/main/java/org/silverpeas/components/blog/dao/PostDAO.java [hunk 2]
```diff
@@ -115,74 +117,31 @@ public static void updateDateEvent(Connection con, String pubId, Date dateEvent)
...
-    }
-    return dateEvents;
+    return select("DISTINCT " + DATE_EVENT)
+        .from(BLOG_POST_TABLE_NAME)
+        .where(INSTANCE_ID_CLAUSE, instanceId)
+        .orderBy(ORDER_BY_DATE)
+        .executeWith(con, r -> new Date(Long.parseLong(r.getString(DATE_EVENT))));
   }
 
   public static Collection<String> getEventsByDates(Connection con, String instanceId,
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply blog/blog-library/src/main/java/org/silverpeas/components/blog/dao/PostDAO.java]: Signature has been renamed.

File: blog/blog-library/src/main/java/org/silverpeas/components/blog/service/BlogFilters.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+   * </p>
+   * @param creatorId identifier of a user.
+   * @return itself.
+   */
+  public BlogFilters withCreatorId(final String creatorId) {
+    this.creatorId = creatorId;
+    return this;
+  }
+
+  public BlogFilters withMaxResultOf(final int maxResult) {
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply blog/blog-library/src/main/java/org/silverpeas/components/blog/service/BlogFilters.java]: OK

File: blog/blog-library/src/main/java/org/silverpeas/components/blog/service/BlogService.java [hunk 4]
```diff
@@ -59,20 +58,22 @@ static BlogService get() {
...
+  Collection<PostDetail> getLastPosts(String instanceId, final BlogFilters filters);
 
-  Collection<PostDetail> getAllValidPosts(String instanceId, int nbReturned);
+  Collection<PostDetail> getAllPosts(String instanceId);
 
-  Date getDateEvent(String pubId);
+  Collection<PostDetail> getLastValidPosts(String instanceId, BlogFilters filters);
 
-  Collection<PostDetail> getPostsByCategory(String categoryId, String instanceId);
+  Collection<PostDetail> getPostsByCategory(String categoryId, String instanceId,
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply blog/blog-library/src/main/java/org/silverpeas/components/blog/service/BlogService.java]: OK

File: blog/blog-library/src/main/java/org/silverpeas/components/blog/service/DefaultBlogService.java [hunk 5]
```diff
@@ -139,12 +137,11 @@ public boolean isRelatedTo(final String instanceId) {
     return instanceId.startsWith("blog");
   }
 
-  @Override
-  public Date getDateEvent(String pubId) {
-    try (Connection con = openConnection()) {
-      return PostDAO.getDateEvent(con, pubId);
+  private Map<String, Date> getDateByEvents(Collection<String> pubIds) {
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply blog/blog-library/src/main/java/org/silverpeas/components/blog/service/DefaultBlogService.java]: Signature has been renamed.

**Pull request merged**